### PR TITLE
Provide a configuration to control the length of the peer field

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.
https://github.com/apache/skywalking/issues/3096

- How to fix?
Add  Peer max length configuration, Used it in the interception of Jedis, Redission, and Lettuce cluster configuration.

if `agent.peer_max_length=5` 
topo like this
image
trace like this
image

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java [hunk 1]
```diff
@@ -93,6 +93,11 @@
          * How depth the agent goes, when log cause exceptions.
          */
         public static int CAUSE_EXCEPTION_DEPTH = 5;
+
+        /**
+         * Peer max length, Used in the interception of Jedis, Redission, and Lettuce cluster configuration.
+         */
+        public static int PEER_MAX_LENGTH = 200;
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java]: Move this to Plugin, this is not a part of core API, right? And don't put plugin name in comments, you wouldn't know when someone add it to somewhere else, considering `PeerUtil` is a core API

**Comment by zhaoyuguang** [comment]
> Is this PR ready? If so, change the title, and don't use issue number as the title, which is hard to read from PR list view.

OK, The PR is ready.

**Comment by wu-sheng** [comment]
@zhaoyuguang Follow the review, please.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java (hunk 1 above)
Comment (zhaoyuguang) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java]: I modified the comments and MD documentation
Moved the `PEER_MAX_LENGTH ` configured location
Modified the name of the class and changed to the enum tool class
Please review code again

**Comment by zhaoyuguang** [comment]
I fix the review issue, Please review code again @wu-sheng

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.util.StringUtil;
+
+/**
+ * @author zhaoyuguang
+ */
+
+public enum PeerFormat {
+
+    INSTANCE;
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java]: There is no need to set in singleton. Just keep it as normal class(new one every time) or static method,

Comment (zhaoyuguang) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java]: Thanks for guidance, this does look better.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java [hunk 3]
```diff
@@ -25,13 +25,11 @@
...
-
-    INSTANCE;
+public class PeerFormat {
 
     private static final String ABBR = "...";
 
-    public String Shorten(String original) {
+    public static String shorten(String original) {
         if (!StringUtil.isEmpty(original) && original.length() > Config.Plugin.PEER_MAX_LENGTH) {
             return original.substring(0, Config.Plugin.PEER_MAX_LENGTH - 3) + ABBR;
```
Comment (kezhenxu94) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java]: What if users set `Config.Plugin.PEER_MAX_LENGTH` to less than `3` (intentionally or by mistake) 😄

Comment (zhaoyuguang) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java]: Can cause unexpected exceptions, the size of the prompt on the document needs to exceed 3 and the judgment of increasing the length of `PEER_MAX_LENGTH ` is less than 3, less than 3 will be invalid and the default value is 200.

Comment (zhaoyuguang) [reply apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/util/PeerFormat.java]: image
I found another way, is this better?

**Pull request merged**